packages/core/admin/admin/src/pages/ProfilePage/index.js

```diff
@@ -73,50 +44,94 @@ const ProfilePage = () => {
   const { lockApp, unlockApp } = useOverlayBlocker();
   const { notifyStatus } = useNotifyAT();
   const { currentTheme, themes: allApplicationThemes, onChangeTheme } = useThemeToggle();
+  const { get, put } = useFetchClient();
+
   useFocusWhenNavigate();
 
-  const { status, data } = useQuery('user', () => fetchUser(), {
-    onSuccess() {
-      notifyStatus(
-        formatMessage({
-          id: 'Settings.profile.form.notify.data.loaded',
-          defaultMessage: 'Your profile data has been loaded',
-        })
-      );
-    },
-    onError() {
-      toggleNotification({
-        type: 'warning',
-        message: { id: 'notification.error', defaultMessage: 'An error occured' },
-      });
-    },
-  });
+  const { isLoading: isLoadingUser, data } = useQuery(
+    'user',
+    async () => {
+      const { data } = await get('/admin/users/me');
 
-  const isLoading = status !== 'success';
+      return data.data;
+    },
+    {
+      onSuccess() {
+        notifyStatus(
+          formatMessage({
+            id: 'Settings.profile.form.notify.data.loaded',
+            defaultMessage: 'Your profile data has been loaded',
+          })
+        );
+      },
+      onError() {
+        toggleNotification({
+          type: 'warning',
+          message: { id: 'notification.error', defaultMessage: 'An error occured' },
+        });
+      },
+    }
+  );
 
-  const submitMutation = useMutation((body) => putUser(body), {
-    async onSuccess(data) {
-      await queryClient.invalidateQueries('user');
+  const { isLoading: isLoadingSSO, data: dataSSO } = useQuery(
+    ['providers', 'isSSOLocked'],
+    async () => {
+      if (window.strapi.isEE) {
+        const {
+          data: { data },
+        } = await get('/admin/providers/isSSOLocked');
 
-      auth.setUserInfo(
-        pick(data, ['email', 'firstname', 'lastname', 'username', 'preferedLanguage'])
-      );
-      const userDisplayName = data.username || getFullName(data.firstname, data.lastname);
-      setUserDisplayName(userDisplayName);
-      changeLocale(data.preferedLanguage);
-      onChangeTheme(data.currentTheme);
-      trackUsage('didChangeMode', { newMode: data.currentTheme });
+        return data;
+      }
 
-      toggleNotification({
-        type: 'success',
-        message: { id: 'notification.success.saved', defaultMessage: 'Saved' },
-      });
+      return {
+        isSSOLocked: false,
+      };
     },
-    onSettled() {
-      unlockApp();
+    {
+      onSuccess() {},
+      onError() {
+        toggleNotification({
+          type: 'warning',
+          message: { id: 'Settings.permissions.users.sso.provider.error' },
+        });
+      },
+    }
+  );
+
+  const isLoading = isLoadingUser || isLoadingSSO;
+
+  const submitMutation = useMutation(
+    async (body) => {
+      const dataToSend = omit(body, ['confirmPassword', 'currentTheme']);
```

Comment:
We're trying not to introduce more `lodash` usage to be honest. What about this:
```suggestion
	const { confirmPassword, currentTheme, ...dataToSend } = body
```

Reply:
I pretty sure those appeared as changes because of the formatter, I don't we want to change anything, also regarding lodash we are using pretty heavily on the backend at least `lodash/fp` am not sure why things should be different on the frontend, regardless it would be nice discussion for a different time